test(std/os/signal): TestNotifyMultipleSignals is flaky under c-archive

## What happened

`TestNotifyMultipleSignals` failed in the Ubuntu primary shard while running the `c-archive` build-mode tests:

```
--- FAIL: TestNotifyMultipleSignals (0.00s)
    signal_test.go:64: First signal = child exited, want SIGWINCH
```

Run: https://github.com/xgo-dev/llgo/actions/runs/30516142565
Job: https://github.com/xgo-dev/llgo/actions/runs/30516142565/job/90786278327

In the same job, the normal llgo test and the c-shared run for `test/std/os/signal` passed. Only the c-archive run failed.

## Cause

The test subscribes the same channel to both `SIGWINCH` and `SIGCHLD`, sends `SIGWINCH` to itself, then assumes the first received signal must be `SIGWINCH`. A subscribed `SIGCHLD` may already be pending and arrive first; ordering between these distinct signals is not guaranteed.

## Proposed fix

Keep the existing timeout, ignore subscribed signals other than `SIGWINCH`, and continue receiving until `SIGWINCH` arrives. This preserves the test's purpose while removing the ordering assumption.

--- a/test/std/os/signal/signal_test.go
+++ b/test/std/os/signal/signal_test.go
@@ -58,13 +58,17 @@ func TestNotifyMultipleSignals(t *testing.T) {
 		t.Fatalf("Signal SIGWINCH error: %v", err)
 	}
 
-	select {
-	case sig := <-c:
-		if sig != syscall.SIGWINCH {
-			t.Errorf("First signal = %v, want SIGWINCH", sig)
+	timeout := time.After(time.Second)
+	for {
+		select {
+		case sig := <-c:
+			// SIGCHLD may arrive first, so wait for the signal sent above.
+			if sig == syscall.SIGWINCH {
+				return
+			}
+		case <-timeout:
+			t.Fatal("Timeout waiting for SIGWINCH")
 		}
-	case <-time.After(time.Second):
-		t.Fatal("Timeout waiting for first signal")
 	}
 }
 

